src/Microsoft.SqlTools.ServiceLayer/LanguageServices/BindingQueue.cs

```diff
@@ -370,49 +370,44 @@ private void ProcessQueue()
                                             }
 
                                             RemoveBindingContext(queueItem.Key);
-                                        }    
+                                        }
                                     }
-                                });
+                                }, new CancellationTokenSource(bindTimeout).Token);
 
-                                Task.Run(() => 
+                                try
                                 {
-                                    try
-                                    {
-                                        // check if the binding tasks completed within the binding timeout                           
-                                        if (bindTask.Wait(bindTimeout))
-                                        {
-                                            queueItem.Result = result;
-                                        }
-                                        else
+                                    await bindTask.ContinueWithOnFaulted(t => Logger.Error("Binding queue threw exception " + t.Exception.ToString()))
```

Comment:
ContinueWithOnFaulted already logs the error so do we really need to log it again? Might be useful to update that to take in a custom error message though since it's pretty generic right now...